src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/IncentiveMappingService.kt

```diff
@@ -32,53 +34,21 @@ class IncentiveMappingService(@Qualifier("mappingApiWebClient") private val webC
         Mono.empty()
       }.awaitSingleOrNull()
 
-  suspend fun createNomisIncentiveMigrationMapping(
-    nomisBookingId: Long,
-    nomisIncentiveSequence: Long,
-    incentiveId: Long,
-    migrationId: String,
-  ) {
-    createNomisIncentiveMapping(
-      nomisBookingId = nomisBookingId,
-      nomisIncentiveSequence = nomisIncentiveSequence,
-      incentiveId = incentiveId,
-      migrationId = migrationId,
-      mappingType = "MIGRATED",
-    )
-  }
-
-  suspend fun createNomisIncentiveSynchronisationMapping(
-    incentiveNomisMapping: IncentiveNomisMapping,
-  ) {
-    createNomisIncentiveMapping(
-      nomisBookingId = incentiveNomisMapping.nomisBookingId,
-      nomisIncentiveSequence = incentiveNomisMapping.nomisIncentiveSequence,
-      incentiveId = incentiveNomisMapping.incentiveId,
-      mappingType = incentiveNomisMapping.mappingType,
-    )
-  }
-
-  private suspend fun createNomisIncentiveMapping(
-    nomisBookingId: Long,
-    nomisIncentiveSequence: Long,
-    incentiveId: Long,
-    mappingType: String,
-    migrationId: String? = null,
-  ) {
-    webClient.post()
+  override suspend fun createMapping(
+    mapping: IncentiveNomisMapping,
+  ): CreateMappingResult {
+    return webClient.post()
       .uri("/mapping/incentives")
       .bodyValue(
-        IncentiveNomisMapping(
-          nomisBookingId = nomisBookingId,
-          nomisIncentiveSequence = nomisIncentiveSequence,
-          incentiveId = incentiveId,
-          label = migrationId,
-          mappingType = mappingType,
-        ),
+        mapping,
       )
       .retrieve()
       .bodyToMono(Unit::class.java)
-      .awaitSingleOrNull()
+      .map { CreateMappingResult() }
+      .onErrorResume(WebClientResponseException.Conflict::class.java) {
+        Mono.just(CreateMappingResult(it.getResponseBodyAs(DuplicateIncentiveErrorResponse::class.java)))
+      }
+      .awaitFirstOrDefault(CreateMappingResult())
   }
```

Comment:
adding the duplicate mapping error (this will not be thrown by the mapping service until the work done at that side)